### PR TITLE
Allow per processor table synchronization.

### PR DESCRIPTION
**Pull request by jmaxson-ms**
Table synchronization at the processor level provides greater control over which columns are required to be updated when applying symbols changes.

Attempting to preserve backcompat by adding new, optional interfaces.

### CODE REVIEW

**Review by trgibeau — APPROVED** [review]

**Review by trgibeau — APPROVED** [review]

File: src/Microsoft.Performance.SDK.Runtime/ApplicationEnvironment.cs [hunk 1]
```diff
@@ -148,5 +149,29 @@ private static MessageBoxIcon MapToImage(MessageType messageType)
...
+        private sealed class NullTableSynchronizer
+            : ITableDataSynchronization
+        {
+            public void SubmitColumnChangeRequest(
+                IEnumerable<Guid> columns,
+                Action onReadyForChange,
+                Action onChangeComplete,
+                bool requestInitialFilterReevaluation = false)
+            {
+                onReadyForChange?.Invoke();
```
Comment (mslukebo) [inline src/Microsoft.Performance.SDK.Runtime/ApplicationEnvironment.cs:162]: according to the spec of this interface, these calls should happen in a lock

Comment (jmaxson-ms) [reply src/Microsoft.Performance.SDK.Runtime/ApplicationEnvironment.cs:162]: Haha, I copied and pasted this from the runtime implementation.

Comment (jmaxson-ms) [reply src/Microsoft.Performance.SDK.Runtime/ApplicationEnvironment.cs:162]: Actually, we shouldn't need the runtime implementation anymore given this change.

Comment (mslukebo) [reply src/Microsoft.Performance.SDK.Runtime/ApplicationEnvironment.cs:162]: Perhaps we should add the lock anyway :)

Comment (jmaxson-ms) [reply src/Microsoft.Performance.SDK.Runtime/ApplicationEnvironment.cs:162]: Do you mean this comment?

`Lock tables affected by the change (based on <paramref name="predicate"/>) and wait for all existing operations on those tables to complete.`

**Review by mslukebo — APPROVED** [review]

**Review by trgibeau — APPROVED** [review]

**Review by mslukebo — APPROVED** [review]

**Pull request merged**